sys/include/net/l2util.h

```diff
@@ -133,6 +134,125 @@ int l2util_ipv6_iid_to_addr(int dev_type, const eui64_t *iid, uint8_t *addr);
 int l2util_ndp_addr_len_from_l2ao(int dev_type,
                                   const ndp_opt_t *opt);
 
+/**
+ * @brief Board-specific function to supply a short address to a netdev.
+ *
+ * @note  Implement this function in your board code if the board
+ *        provides the means to supply a unique address to a netdev.
+ *
+ * @warning Don't call this function directly, use @ref l2util_generate_short_addr() instead.
+ *
+ * @param[in] driver    The driver name of the netdev
```

Comment:
Please relight my memory: What is the driver name of the netdev?

Reply:
The idea was that a EUI would be assigned to a certain device/driver.
E.g. both `samr21-xpro` and `same54-xpro` contain a EUI-64 resp. EUI-48.
On `samr21-xpro` that should only be used for the first device of the `at86rf2xx` driver…

It's not pretty but so far we haven't come up with a better solution to assign IDs to netdevs.

Reply:
> It's not pretty but so far we haven't come up with a better solution to assign IDs to netdevs.

Mhhhh its not just not pretty, it can be very costly. Both the string will cost memory, and string operations aren't the most efficient either. :-/

Reply:
Would it make sense to use [`NETDEV_TYPE_`](https://doc.riot-os.org/group__drivers__netdev__api.html) here?

Reply:
If the `netdev` allocations would be in an [xfa](https://github.com/RIOT-OS/RIOT/pull/9105) this would be simpler :-/.

Reply:
> Would it make sense to use `NETDEV_TYPE_` here?

Not really - the idea was that if you e.g. attach a mrf24j40 to a `samr21-xpro`, thet at86rf2xx radio would still get the MAC address that's printed on the board.

A better option would be if netdevs had IDs, but right now they just have a PID and that changes depending on what modules are used.

Reply:
> A better option would be if netdevs had IDs, but right now they just have a PID and that changes depending on what modules are used.

Where do they have a PID? `gnrc_netif`s have a PID, but those are not `netdev`s ;-).

Reply:
Oh yea, got confused there - well then afaik there is no way to uniquely identify a `netdev` ;-)

Reply:
> [1] I'd be fine with just handing out UIDs in a first-come first-serve basis.
That's the common case anyway and it would make things much simpler.

Mh, yeah, why don't we ask of a `netdev` driver to provide some globally unique numeric value (e.g. `AT86RF2XX_DEVUID`) in their header. If a board then implements `board_get_…`, it can use that value to identify the devices.

Reply:
Then if you have more than one `netdev`s with the same driver it's still first-come first-serve (as with the string based approach)

Reply:
What do you mean by first-come first serve? I thought `idx` is in line with the `params` array (which given that at the moment `auto_init` calls the setup functions in that order, should be the case).

Reply:
> even if locally in the driver we don't know the position

I can think of three options:

 - `at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)` -> `at86rf2xx_setup(at86rf2xx_t *dev, uint8_t idx)` - a bit tedious but probably the cleanest solution
- `(params - &at86rf2xx_params) / sizeof(*params)` to get the index (urgh)
- introduce a local counter in the driver (and waste a precious byte of RAM)

Reply:
Why not provide a board-unique ID with `params` btw?

Reply:
For this we would have the migration benefit, that this could be done driver by driver.

Reply:
> Why not provide a board-unique ID with `params` btw?

- we have to touch every board
- we waste RAM and ROM for information that is already implicitly available

Reply:
>     * we waste RAM and ROM for information that is already implicitly available

I thought it is not. If it is why use the (more RAM heavy) name?



>     * we have to touch every board

If we don't use the current approach, I believe we don't have. Also: only every board that provides a netdev `params`.

Reply:
> I thought it is not. If it is why use the (more RAM heavy) name?

Well we just have to touch every netdev driver (to do the index -> params mapping inside the driver instead of auto_init). I just want to be sure that's the right way to go before I shave that Yak 😉

Reply:
> I just want to be sure that's the right way to go before I shave that Yak wink

See https://github.com/RIOT-OS/RIOT/pull/12641#issuecomment-580240006. As far as I understand `board.c` might not exist anymore in the near future :-/.